### DISCUSSION TITLE
Where are the API docs for the Directus PHP classes?

### DISCUSSION THREAD

**Discussion by weaversam8** [directus/directus#2717] [Archive]
I was reading [the Directus docs about building Extensions](https://docs.directus.io/extensions/#architecture-explanation), specifically about building [Custom Endpoints](https://docs.directus.io/extensions/custom-endpoints.html).

Some [example code](https://docs.directus.io/extensions/custom-endpoints.html#creating-custom-endpoints) on this page references that custom code can use the Directus code. This code in particular uses the `ItemsService` class, but there seem to be a lot of useful classes in the `src/core/Directus` directory.

Does documentation exist for these classes? If so, where is it?

**Comment by rijkvanzanten** [comment]
I don't think we have a full docs for those somewhere centralized.. Those were build over time by a wide range of open source contributors. We're actually in the middle of migrating all of that over to Laravel (over on https://github.com/directus/api-next), which is also the perfect time to document those things (@WoLfulus )

**Comment by WoLfulus** [comment]
The new Laravel version has a very straightforward and clean API to interact with Directus. Once alpha is out I'll start working on the documentation. Users will likely use Laravel's documentation most of the time, but I'll add some concepts and guidelines to our documentation as well.
